server/src/modules/task/task.service.ts

```diff
@@ -1,17 +1,42 @@
-import { Injectable } from '@nestjs/common';
-import { User } from '../user/user.model';
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { Task } from './task.model';
+import { InjectModel } from '@nestjs/sequelize';
+import { TaskDTO as TaskInterface } from './task.interface';
 
 @Injectable()
 export class TaskService {
-  async createUser(createUserDto: any): Promise<User> {
-    const user = new User();
-    user.firstName = createUserDto.username;
-    user.email = createUserDto.email;
+  constructor(
+    @InjectModel(Task)
+    private readonly taskModel: typeof Task,
+  ) {}
 
-    return user.save();
+  async createTask(taskData: Task): Promise<Task> {
+    return this.taskModel.create(taskData);
   }
 
-  async findAllUsers(): Promise<User[]> {
-    return User.findAll();
+  async getAllTask(): Promise<TaskInterface[]> {
```

Comment:
One of these returns `Task` and the other `TaskInterface`: is that intended? (Same question on other methods in this controller)